### PR TITLE
Optimized group membership resolution

### PR DESCRIPTION
**Pull request by manojampalam**
Current group membership resolution though very effective, is very slow. In a typical domain joined enterprise machine, adding a simple entry like the following in sshd_config
AllowGroups administrators
will incur a long delay in remote session establishment as sshd tried to pull all groups associated with the domain user.

Changes in this PR optimize the general case scenarios where no wild cards are in use. Specifically rules like this are processed promptly:

AllowGroups group1, group2, group3 //with no wild cards
Match Group group1 //single group with no negation and wild cards

Optimization is done by resolve the groupname in rule immediately to SID and checking its membership against user token. Enumerating the entire group membership is done on a lazy on-demand basis.

### CODE REVIEW

**Comment by manojampalam** [comment]
@nomorefood FYI

**Comment by manojampalam** [comment]
@nomorefood please review win32_groupaccess.c when you have a chance. Beyond the optimization
The are 2 functional changes 
 - removed domain prefix for builtin groups
 - removed domain prefix'ed versions of local groups since we are strictly following the convention that local principals shouldn't have any domain qualification.

**Comment by NoMoreFood** [comment]
@manojampalam Sounds good. I'll try to take a look tonight.

File: contrib/win32/openssh/sshd.vcxproj [hunk 1]
```diff
@@ -1,473 +1,473 @@
...
-    <ClCompile Include="$(OpenSSH-Src-Path)contrib\win32\win32compat\wmain_sshd.c" />
-    <ClCompile Include="$(OpenSSH-Src-Path)contrib\win32\win32compat\spawn-ext.c" />
-  </ItemGroup>
-  <ItemGroup>
-    <ResourceCompile Include="version.rc" />
-  </ItemGroup>
-  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
-  <ImportGroup Label="ExtensionTargets">
-  </ImportGroup>
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (bagajjal) [inline contrib/win32/openssh/sshd.vcxproj]: Looks like CRLF issue.. please take care of this.

Comment (manojampalam) [reply contrib/win32/openssh/sshd.vcxproj]: done.

**Comment by manojampalam** [comment]
@NoMoreFood I'll have to merge these changes to unblock a couple of other PRs in my pipeline. But please do review groupaccess* changes.

**Comment by NoMoreFood** [comment]
@manojampalam The changes look good to me.

**Pull request merged**